firewood/src/merkle/parallel.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+//use std::collections::HashMap;
+use std::sync::{mpsc, Arc, OnceLock};
+
+//use firewood_storage::{
+//    BranchNode, Child, FileBacked, FileIoError, ImmutableProposal, NibblesIterator, Node,
+//    NodeStore, Parentable, RootReader,
+//};
+
+use firewood_storage::{
+    FileBacked, FileIoError, ImmutableProposal, Node,
+    NodeStore, Parentable,
+};
+
+use rayon::{ThreadPool, ThreadPoolBuilder};
+//use sha2::digest::crypto_common::KeyInit;
+
+//use crate::db;
+use crate::merkle::{Key, Merkle, Value};
+use crate::v2::api::KeyValuePairIter;
+
+/*
+trait DbExt {
+    fn propose_sync_parallel(
+        &self,
+        batch: impl IntoIterator<IntoIter: KeyValuePairIter>,
+    ) -> Result<Proposal<'_>, FileIoError>;
+}
+*/
+
+#[derive(Debug)]
+struct Worker {
+    sender: mpsc::Sender<Request>,
+}
+
+/// A request to the worker.
+#[derive(Debug)]
+enum Request {
+    Insert { key: Key, value: Value },
+    Delete { key: Key },
+    DeleteRange { prefix: Key },
+    Done,
+}
+
+#[derive(Debug)]
+enum Response {
+    // return the new root of the subtrie at the given nibble
+    //Root(u8, Option<Node>),
+    Root(Option<Node>),
+    Error(FileIoError),
+}
+
+
+static THREADPOOL: OnceLock<ThreadPool> = OnceLock::new();
```

Comment:
I'm not sure if this is the right approach here - I personally would prefer just having an async function so that the caller can decide the runtime to use and can share thread with other code instead of spawning new threads.